src/ControlSystem/DataVectorHelpers.hpp

```diff
@@ -0,0 +1,23 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+
+/// \ingroup ControlSystemGroup
+/// Convert a std::array to a DataVector. Currently only instantiated for N=3.
+template <size_t N>
+DataVector array_to_datavector(const std::array<double, N>& arr);
+
+/// \ingroup ControlSystemGroup
+/// Compute the dot product between two DataVectors.
+double dot(const DataVector& a, const DataVector& b);
+
+/// \ingroup ControlSystemGroup
+/// Compute the cross product between two DataVectors. (each DataVector must
+/// have exactly 3 components)
+DataVector cross(const DataVector& a, const DataVector& b);
```

Comment:
This exists in blaze, but is not currently enabled for DataVector.  Add this to DataVector.hpp:
```c++
template <>
struct blaze::CrossTrait<DataVector, DataVector> {
  using Type = DataVector;
};
```

Reply:
Should I add it in DataVector.hpp or VectorImpl.hpp as
```
BLAZE_TRAIT_SPECIALIZE_BINARY_TRAIT(VECTOR_TYPE, CrossTrait);
```
with all the other traits?

Reply:
I don't think it makes sense to do cross products with the more specialized vector types.  Using the macro is better than writing out the struct explicitly, though.